core/src/test/scala/com/salesforce/op/stages/impl/feature/ScalerTransformerTest.scala

```diff
@@ -0,0 +1,84 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.stages.impl.feature
+
+import com.salesforce.op.OpWorkflow
+import com.salesforce.op.features.{Feature, FeatureLike}
+import com.salesforce.op.features.types.Real
+import com.salesforce.op.utils.json.JsonUtils
+import com.salesforce.op.test.{TestFeatureBuilder, TestSparkContext}
+import org.joda.time.DateTime
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.FlatSpec
+
+@RunWith(classOf[JUnitRunner])
+class  ScalerTransformerTest extends FlatSpec with TestSparkContext {
+  val (testData, inA) = TestFeatureBuilder("inA", Seq[(Real)](Real(4.0), Real(1.0), Real(0.0)))
+  val scalingType = "scalingType"
+  val scalingArgs = "scalingArgs"
+
+  Spec[ScalerTransformer[_, _]] should "Properly linearly scale numeric fields and serialize the transformer" in {
+    val linearScaler = new ScalerTransformer[Real, Real](
+      scalingType = ScalingType.Linear,
+      scalingArgs = LinearScalerArgs(slope = 2.0, intercept = 1.0)
+    ).setInput(inA.asInstanceOf[Feature[Real]])
+    val vec: FeatureLike[Real] = linearScaler.getOutput()
+    val wfModel = new OpWorkflow().setResultFeatures(vec).setInputDataset(testData).train()
+    val saveModelPath = tempDir.toPath.toString + "linearScalerTest" + DateTime.now().getMillis.toString
+    wfModel.save(saveModelPath)
+    val data = wfModel.score()
+    val actual = data.collect()
+    actual.map(_.getAs[Double](1)) shouldEqual Array(9.0, 3.0, 1.0)
+    val metadata = data.schema(vec.name).metadata
+    metadata.getString(scalingType) shouldEqual ScalingType.Linear.entryName
+    val args: LinearScalerArgs = JsonUtils.fromString[LinearScalerArgs](metadata.getString(scalingArgs)).get
+    args.intercept shouldEqual 1.0
+    args.slope shouldEqual 2.0
+  }
+
+  it should "Properly log scale numeric fields and serialize the transformer" in {
+    val predScaler = new ScalerTransformer[Real, Real](
+      scalingType = ScalingType.Logarithmic,
+      scalingArgs = EmptyArgs()
+    ).setInput(inA.asInstanceOf[Feature[Real]])
+    val vec: FeatureLike[Real] = predScaler.getOutput()
+    val wfModel = new OpWorkflow().setResultFeatures(vec).setInputDataset(testData).train()
```

Comment:
so you actually dont need a workflow to test a single transformer. see how you did it here for the date circle transformer for example ;) - https://github.com/salesforce/TransmogrifAI/blob/master/core/src/test/scala/com/salesforce/op/stages/impl/feature/DateToUnitCircleTransformerTest.scala#L84

Reply:
(same applies in other test cases)